Editor/EditorCore/MainToolbar.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ProBuilder.UI
+{
+    class ToolbarMenuItem : VisualElement
+    {
+        public new class UxmlFactory : UxmlFactory<ToolbarMenuItem, UxmlTraits> { }
+        public MenuAction action;
+        public bool iconMode;
+
+        public void RefreshContents()
```

Comment:
I have an error in the editor when switching from text to icon mode (2023.3)
![image](https://github.com/Unity-Technologies/com.unity.probuilder/assets/66317117/8f41cc6f-a6a0-41eb-b17b-fa340e114161)

Reply:
There is also a lag when all the content is switching from vertical to horizontal mode (and the other direction too)
![probuilder-layoutchanged](https://github.com/Unity-Technologies/com.unity.probuilder/assets/66317117/0d42048a-a4e2-4da1-8a93-eab4ae6d0936)

Reply:
The lag is expected, it's rebuilding the visual element tree.